### PR TITLE
tests/bluetooth/tester: Add support for auto connection establishment

### PR DESCRIPTION
**Pull request by sjanc**
autopts was updated to properly require support for Accept Filter List
in Auto Connection Establishment Procedure related tests. This patch
enabled support for it and adds required BTP support.

This was affecting following qualification test cases:
GAP/CONN/ACEP/BV-03-C
GAP/CONN/ACEP/BV-04-C

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>

### CODE REVIEW

File: tests/bluetooth/tester/src/gap.c [hunk 1]
```diff
@@ -1176,6 +1190,33 @@ static void set_oob_legacy_data(const uint8_t *data, uint16_t len)
 		   CONTROLLER_INDEX, BTP_STATUS_SUCCESS);
 }
 
+static void set_filter_list(const uint8_t *data, uint16_t len)
+{
+	const struct gap_set_filter_list *cmd = (const void *) data;
+	uint8_t status = BTP_STATUS_FAILED;
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/gap.c:1196]: Why not set the status in the error branches, same way as you do in the `connect()` function?

Comment (sjanc) [reply tests/bluetooth/tester/src/gap.c]: updated, status is now set in error branches

File: tests/bluetooth/tester/src/gap.c [hunk 2]
```diff
@@ -808,19 +809,32 @@ static void stop_discovery(const uint8_t *data, uint16_t len)
 
 static void connect(const uint8_t *data, uint16_t len)
 {
-	struct bt_conn *conn;
+	const bt_addr_le_t *addr = (const bt_addr_le_t *) data;
```
Comment (Thalley) [inline tests/bluetooth/tester/src/gap.c:812]:
```suggestion
	const bt_addr_le_t *addr = (const bt_addr_le_t *)data;
```

**Review by Thalley — DISMISSED** [review]
Minor nitpick comment, but LGTM

**Review by Thalley — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**